staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -315,9 +318,19 @@ func cleanVerb(verb string, request *http.Request) string {
 	if verb == "WATCHLIST" {
 		reportedVerb = "WATCH"
 	}
+	if verb == "PATCH" && request.Header.Get("Content-Type") == string(types.ApplyPatchType) {
+		reportedVerb = "APPLY"
+	}
 	return reportedVerb
 }
 
+func cleanDryRun(dryRun []string) string {
+	if err := validation.ValidateDryRun(nil, dryRun); err != nil {
+		return "invalid"
+	}
+	return strings.Join(dryRun, ",")
```

Comment:
there are still an infinite number of strings that would pass validation, need to dedupe and sort to make a canonical metrics key

Reply:
Well, unfortunately this is still not super well defined. Does the order matter or doesn't it?

Reply:
it does not matter, dryRun is a set